IntGeVaP/IntGeVaP-C.R

```diff
@@ -0,0 +1,420 @@
+              help = "variant depth filter"),
+  make_option(c("--variant_af"), type = "numeric", default = .2,
+              help = "variant AF cut-off"),
+  make_option(c("--summary_level"), type = "character", default = "F",
+              help = "summary_level T/F"),
+  make_option(c("--output"), type = "character", default = "out",
+              help = "output name")) 
+
+opt <- parse_args(OptionParser(option_list = option_list))
+
+## get input files from parameters (reqd)
+input_clinVar_file  <-  opt$vcf
+input_intervar_file <- opt$intervar
+input_autopvs1_file <- opt$autopvs1
+clinvar_ver <- opt$clinvar
+sample_name <- opt$sample_name
+
+## filters for gnomAD
+filter_gnomad_var    <- opt$gnomad_variable
+filter_variant_depth <- opt$variant_depth
+filter_variant_af    <- opt$variant_af
+
+## output files
+output_tab_file <- file.path(analysis_dir, paste0(sample_name, "_annotations_report.tsv")) 
+output_tab_abr_file  <- file.path(analysis_dir, paste0(sample_name,"_annotations_report.abridged.tsv"))
+output_tab_dev_file  <- file.path(analysis_dir, paste0(sample_name,"_annotations_report.abridged.dev.tsv"))
+
+## for testing purposes only ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ###
+#input_clinVar_file  <- file.path(input_dir,"../clinvar.vcf.gz")
+input_intervar_file <- file.path(input_dir,"7aefaa8b-3dbe-4fe2-afb1-6f95d132ec87.hg38_multianno.chr20.txt.intervar")
+input_multianno_file <- file.path(input_dir, "7aefaa8b-3dbe-4fe2-afb1-6f95d132ec87.hg38_multianno.chr20.vcf")
+input_vcf_file      <- file.path(input_dir,"f766f75d-0917-465d-8005-72fddeb8ec6b.single.vqsr.filtered.vep_105.chr20.vcf")
+input_autopvs1_file <- file.path(input_dir,"7aefaa8b-3dbe-4fe2-afb1-6f95d132ec87.autopvs1.chr20.tsv")
+input_submission_file <- file.path(input_dir,"variant_summary.txt")
+input_summary_submission_file <- file.path(input_dir,"submission_summary.txt")
+
+sample_name <-  "test"
+gnomad_variable <- "Freq_gnomAD_genome_ALL"
+gnomad_af <- 0.001
+filter_variant_depth = 15
+## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ###
+
+## allocate more memory capacity
+Sys.setenv("VROOM_CONNECTION_SIZE" = 131072 * 2)
+
+## function for filtering
+##gnomAD, variant af and depth filtering 
+gnomad_filtering <- function(clinvar_vcf) {
+  clinvar_vcf <- clinvar_vcf %>% 
+                  mutate(variant_depth = if_else( as.integer( str_match(INFO, "DP\\=(\\d+)")[, 2])  > filter_variant_depth, "PASS","FAIL")) %>% 
+                  mutate(gnomad_af     = if_else( as.numeric( str_match(INFO, "gnomad_3_1_1_AF_non_cancer\\=(0\\.\\d+)\\;")[,2])  > filter_variant_af, "PASS","FAIL")) %>% 
+                  mutate(variant_af    = if_else(as.integer(str_match(Sample, ":(\\d+)\\,(\\d+)") [,3]) / ( (as.integer(str_match(Sample, ":(\\d+)\\,(\\d+)") [,2]) ) + as.integer(str_match(Sample, ":(\\d+)\\,(\\d+)") [,3] )) > filter_variant_af, "PASS", "FAIL"))
+  return(clinvar_vcf)
+}
```

Comment:
later, we should pull out any functions and place in a util folder, then source them

Reply:
agreed